pkg/daemon/podprobe/pod_probe_controller.go

```diff
@@ -407,6 +411,25 @@ func (c *Controller) fetchLatestPodContainer(podUID, name string) (*runtimeapi.C
 	return containerStatus, err
 }
 
+func (c *Controller) fetchPodIp(podUID, name, namespace string) (string, error) {
```

Comment:
Our internal cluster is quite large and there are performance issues if Daemon needs to watch Pod resources, so it is recommended that Pod IP can be passed down directly from the NodePodProbeSpec resource.